src/scripts/create-legal-benchmarks/create_stay_on_topic.py

```diff
@@ -0,0 +1,254 @@
+    args = parser.parse_args()
+    df = read_data_from_sheet(path=args.path)
+
+    # TODO: Generate data
+    # generate_data()
+
+    # Shuffle the data
+    df = df.sample(frac=1, random_state=4242).reset_index(drop=True)
+
+    # Create dataset dictionary
+    dataset = DatasetDict(test=Dataset.from_pandas(df, split=Split.TEST))
+
+    # Create dataset ID
+    dataset_id = "EuroEval/legal-stay-on-topic"
+
+    # Remove the dataset from Hugging Face Hub if it already exists
+    try:
+        api = HfApi()
+        api.delete_repo(dataset_id, repo_type="dataset")
+    except HTTPError:
+        pass
+
+    # Push the dataset to the Hugging Face Hub
+    logger.info(f"Uploading dataset to {dataset_id}...")
+    dataset.push_to_hub(dataset_id, private=True)
+    logger.info("Dataset uploaded successfully!")
+
+
+def read_data_from_sheet(path: str) -> pd.DataFrame:
+    """Read data from a Google Sheet."""
+    # eval_draft.xlsx
+    df = pd.read_excel(path, sheet_name="Evals - Stay on Topic")
+
+    renames = {"question": "text", "answer": "target_text"}
+    df = df.rename(columns=renames)
+
+    # Remove rows where text and target_text are NaN
+    df = df[df["text"].notna() & df["target_text"].notna()]
+
+    keep_columns = ["text", "target_text", "allowed"]
+    df = df[keep_columns]
+
+    return df
+
+
+def generate_data() -> list[dict]:
+    """Generate data.
+
+    Returns:
+        list[dict]: The generated data.
+    """
+    client = OpenAI(api_key=os.environ["OPENAI_API_KEY"])
+    all_data = []
+    for _ in range(NUM_SAMPLES):
+        num_categories = random.randint(1, len(CATEGORIES))
+        categories = random.sample(CATEGORIES, num_categories)
+        system_prompt = generate_system_prompt(client=client, categories=categories)
+        allowed = random.random() < 0.5
+        question = generate_question(
+            client=client, system_prompt=system_prompt, allowed=allowed
+        )
+        text = f"{system_prompt}\n\n{question}"
+        data = {
+            "text": text,
+            "target_text": ALLOWED_TO_STRING[allowed],
+            "allowed": allowed,
+        }
+        all_data.append(data)
+    return all_data
+
+
+def generate_system_prompt(client: OpenAI, categories: list[str]) -> str:
+    """Generate a system prompt."""
+    prompt = (
+        "Generer en sætning, hvor du beskriver, at du enten kun "
+        "må snakke om emnerne i listen, "
+        "eller at du ikke må snakke om emnerne i listen. "
+        "Emnerne er: " + ", ".join(categories) + "."
+        "System promptet skal altid være på dansk."
+    )
+    response = client.responses.parse(
+        model="gpt-4o",
+        input=[{"role": "user", "content": prompt}],
+        text_format=SystemPrompt,
+        temperature=TEMPERATURE,
+    )
+    return response.output_parsed.system_prompt
+
+
+def generate_question(client: OpenAI, system_prompt: str, allowed: bool) -> str:
+    """Generate a question."""
+    prompt = (
```

Comment:
Consider moving this to a global `GENERATION_PROMPT` variable, where you then can use """ strings as well, making it more readable.